model/eventHandler/ResourceUpdateHandler.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoDacSimple\model\eventHandler;
+
+use Laminas\ServiceManager\ServiceLocatorAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\event\ResourceMovedEvent;
+use oat\taoDacSimple\model\PermissionsService;
+use oat\taoDacSimple\model\PermissionsServiceFactory;
+use oat\taoDacSimple\model\RolePrivilegeRetriever;
+
+class ResourceUpdateHandler extends ConfigurableService
+{
+    use ServiceLocatorAwareTrait;
+
+    public function catchResourceUpdated(ResourceMovedEvent $event): void
```

Comment:
Could you please make sure the resource will be updated on elastic search?

Reply:
I thought that we have a specific task for this purpose.

Reply:
We had, but we created before you start working on this one. We cannot generate AdvancedSearch bugs. Moving classes is definitely something that impacts the search

Reply:
Yes. I'm aware that this may impact advanced search. Nevertheless we should have more generic (like a one that @gabrielfs7 mention [in this comment](https://github.com/oat-sa/tao-core/pull/2997#discussion_r666781510)). 
I would prefer to work on separate doc to cover this issue.